src/middleware/submissionHandler.js

```diff
@@ -383,6 +383,56 @@ module.exports = function(router, resourceName, resourceId) {
       done();
     };
 
+    /**
+     * Set parenthood of child forms if not already done
+     *
+     * @param req
+     * @param res
+     * @param done
+     */
+    var setChildFormParenthood = function(req, res, done) {
+      if (req.method !== 'POST' && req.method !== 'PUT') {
+        return done();
+      }
+
+      _.each(req.flattenedComponents, function(component, key) {
```

Comment:
I like what you are doing here, but this would probably be better positioned in the fieldAction handlers.  https://github.com/formio/formio/tree/master/src/actions/fields

You would just create one called ```form.js```. These get executed in a very similar way that you are already doing here so you would get some performance boost. Here is where they are triggered. https://github.com/formio/formio/blob/master/src/middleware/submissionHandler.js#L353

This basically does what you are doing, you would just create an ```afterPost``` and ```afterGet``` method and implement this feature. Similar to what the Email validation handler does. https://github.com/formio/formio/blob/master/src/actions/fields/email.js. You would just not need to wait on the update to be made, but call the ```next``` handler immediately.

Thanks!